Subtitles continuing

miigotu opened the pull request:
I think this is done.

miigotu commented:
@neoatomic @fernandog

miigotu commented:
If you change any flags, make a PR to the subliminal-continuing branch.

neoatomic commented:
@miigotu I found this collection of alpha3 named flags, but they are in *.gif and 16x11. However would be a quick fix. https://github.com/adamoliver/Country-Flags-ISO-3

miigotu commented:
@neoatomic GREAT FIND! If I can change the code to use gif, maybe we can git subtree that.

fernandog commented:
in the gif there's the bra.png (brazilian portuguese) BUT in the lang codes there isn't.

http://www.localeplanet.com/icu/pt-BR/index.html
http://www.localeplanet.com/icu/pt-PT/index.html
both use POR.
xbmc/kodi uses POR too

opensubtitles use POB to differentiate

miigotu commented:
@fernandog `bra` is not portuguese. 

`lib/babelfish/data/iso-639-3.tab`
`bra     bra     bra             I       L       Braj`
`por     por     por     pt      I       L       Portuguese`

fernandog commented:
But the flag is from brazil

miigotu commented:
Eh, @neoatomic too bad those flags are ISO 3166-1 alpha-3 not ISO 639-3 alpha-3

miigotu commented:
@fernandog @neoatomic I think the remaining alpha2 flags are just country flags. Where do we even use those?

fernandog commented:
Sorry. Dont use SR subtitles.
history page maybe?

miigotu commented:
@fernandog imdb info =P...

@neoatomic can you test this now before I merge? I have tested with single and multi subs, 1 and many languages. You probably won't have any ? icons now too.

I still have at least 2 small things I would like to improve here, but it is good enough for now I think.

neoatomic commented:
@miigotu If you have tested yourself than just merge it. (That makes it also more easy for me to test.)

neoatomic commented:
@miigotu , Did some testing. 

First rescanned the show-folders. No errors and the Dutch flags are show for the (previous) downloaded subs. Than i removed one sub and clicked on "cc" for a manual search. And SR downloaded the sub. However the notification showed wrong. : 
![2](https://cloud.githubusercontent.com/assets/7928052/8523119/b98295f8-23f3-11e5-984e-d237cde090c5.png)

(In the past "cc" was not visible if a sub was detected in the folder. So not sure if that is working correctly.?) 
Then i refreshed the page and the episode showed the characters "nl" instead of the image.
![1](https://cloud.githubusercontent.com/assets/7928052/8523156/e4a47de6-23f3-11e5-87e4-b2c13c793d33.png)

Then i went to the history page, but no flags for the newly downloaded or previous downloaded where shown. : 

![3](https://cloud.githubusercontent.com/assets/7928052/8523219/401794f6-23f4-11e5-955d-5bc95e629986.png)

The flag is : http://192.168.2.3:8899/images/flags/unknown.png
The (emty) link is : http://192.168.2.3:8899/images/flags/nl.png

It clearly requests the alpha2 flags not the alpha3. (Maybe my DB still had the old download data with the alpha2 results.?)
Also in the old situation the flag showed in the "action" column, but in the new in the provider column.

Then i switched the view of the history page from "detailed" to "compact" and got an error. : 
(And as i set it to compact i cant open history now. Probably need to manually set that back in the config.ini.)

```
2015-07-06 15:26:39 DEBUG    Thread-14 :: 121361: Downloading subtitles for S01E01
2015-07-06 15:26:53 DEBUG    Thread-14 :: 121361: Saving episode details to database
2015-07-06 15:26:53 DEBUG    Thread-14 :: STATUS IS 25604
2015-07-06 15:26:53 DEBUG    Thread-14 :: 121361: Downloaded Dutch subtitles for S01E01
2015-07-06 15:26:54 DEBUG    Thread-14 :: Parsed /volume1/video/=_Series_=/Game Of Thrones/Game of Thrones S01E01.1080p.BluRay.x264.anoXmous_ into Game of Thrones - S1E1 [ANIME VER: -1] [ABD: False] [ANIME: False] [whichReg: ['standard']]
2015-07-06 15:26:54 DEBUG    Thread-14 :: Episode has no release name, replacing it with a generic one: %SN - %Sx%0E - %EN
2015-07-06 15:26:54 DEBUG    Thread-14 :: formatting pattern: %SN - %Sx%0E - %EN -> Game of Thrones - 1x01 - Winter is Coming
2015-07-06 15:26:54 DEBUG    Thread-14 :: Notification for NMJ scan update not enabled, skipping this notification
2015-07-06 15:26:54 DEBUG    Thread-14 :: history.logSubtitle Dutch
2015-07-06 15:37:52 ERROR    Thread-22 :: Failed doing webui callback: Traceback (most recent call last):
  File "/volume1/@appstore/********/var/SickRage/sickbeard/webserve.py", line 259, in async_call
    result = function(**kwargs)
  File "/volume1/@appstore/********/var/SickRage/sickbeard/webserve.py", line 3588, in index
    return t.respond()
  File "/usr/local/********/var/cache/cheetah/_usr_local_********_var_SickRage_gui_slick_interfaces_default_history_tmpl.py", line 545, in respond
    _v = VFN(VFFSL(SL,"action",True)['provider'],"png",True) # u"$action['provider'].png" on line 225, col 43
NotFound: cannot find 'png' while searching for 'png'
Traceback (most recent call last):
  File "/volume1/@appstore/********/var/SickRage/sickbeard/webserve.py", line 259, in async_call
    result = function(**kwargs)
  File "/volume1/@appstore/********/var/SickRage/sickbeard/webserve.py", line 3588, in index
    return t.respond()
  File "/usr/local/********/var/cache/cheetah/_usr_local_********_var_SickRage_gui_slick_interfaces_default_history_tmpl.py", line 545, in respond
    _v = VFN(VFFSL(SL,"action",True)['provider'],"png",True) # u"$action['provider'].png" on line 225, col 43
NotFound: cannot find 'png' while searching for 'png'
2015-07-06 15:37:52 DEBUG    TORNADO :: Failed doing webui request "index": Traceback (most recent call last):
  File "/volume1/@appstore/********/var/SickRage/sickbeard/webserve.py", line 244, in get
    results = yield self.async_call(method)
  File "/volume1/@appstore/********/var/SickRage/tornado/gen.py", line 807, in run
    value = future.result()
  File "/usr/local/********/var/SickRage/lib/concurrent/futures/_base.py", line 400, in result
    return self.__get_result()
  File "/usr/local/********/var/SickRage/lib/concurrent/futures/_base.py", line 359, in __get_result
    reraise(self._exception, self._traceback)
  File "/usr/local/********/var/SickRage/lib/concurrent/futures/_compat.py", line 107, in reraise
    exec('raise exc_type, exc_value, traceback', {}, locals_)
  File "/usr/local/********/var/SickRage/lib/concurrent/futures/thread.py", line 61, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/volume1/@appstore/********/var/SickRage/sickbeard/webserve.py", line 259, in async_call
    result = function(**kwargs)
  File "/volume1/@appstore/********/var/SickRage/sickbeard/webserve.py", line 3588, in index
    return t.respond()
  File "/usr/local/********/var/cache/cheetah/_usr_local_********_var_SickRage_gui_slick_interfaces_default_history_tmpl.py", line 545, in respond
    _v = VFN(VFFSL(SL,"action",True)['provider'],"png",True) # u"$action['provider'].png" on line 225, col 43
NotFound: cannot find 'png' while searching for 'png'
```

hlavicka commented:
Hi @miigotu 

I also tested the last update and it does not download the subtitles as well as the standalone subliminal (0.8.0-dev). 

When I tried to download subtitles (Czech, English) manually using "cc" icon only English subtitle was downloaded. I have all the providers enabled. This was in the log:

```
2015-07-06 17:06:26 DEBUG    Thread-28 :: 289590: Downloading subtitles for S01E02
2015-07-06 17:06:34 DEBUG    Thread-28 :: 289590: No subtitles found for S01E02 on any provider
2015-07-06 17:13:00 DEBUG    Thread-28 :: 289590: Downloading subtitles for S01E02
2015-07-06 17:13:52 DEBUG    Thread-28 :: 289590: Saving episode details to database
2015-07-06 17:13:53 DEBUG    Thread-28 :: STATUS IS 404
2015-07-06 17:13:54 DEBUG    Thread-28 :: 289590: Downloaded English subtitles for S01E02
2015-07-06 17:13:55 DEBUG    Thread-28 :: Parsed /i-data/md0/video/TVShows/Mr. Robot/Mr. Robot - 1x02 - EPS1.1_ONES-AND-ZER0ES.MPEG into Mr. Robot - S1E2 [ANIME VER: -1] [GROU
2015-07-06 17:13:55 DEBUG    Thread-28 :: Parsed Mr.Robot.S01E02.720p.HDTV.x264-KILLERS into Mr Robot - S1E2 [ANIME VER: -1] [GROUP: KILLERS] [ABD: False] [ANIME: False] [whic
2015-07-06 17:13:55 DEBUG    Thread-28 :: formatting pattern: %SN - %Sx%0E - %EN -> Mr. Robot - 1x02 - eps1.1_ones-and-zer0es.mpeg
2015-07-06 17:13:55 DEBUG    Thread-28 :: Notification for NMJ scan update not enabled, skipping this notification
2015-07-06 17:13:55 DEBUG    Thread-28 :: NMA: Sending notice with details: event="Subtitle Download Finished", message="Mr. Robot - 1x02 - eps1.1_ones-and-zer0es.mpeg: Englis
2015-07-06 17:13:56 INFO     Thread-28 :: NMA: Notification sent to NotifyMyAndroid
2015-07-06 17:13:56 DEBUG    Thread-28 :: history.logSubtitle English
```
However I was able to download the Czech subtitle using subliminal cli:
```
root@NSA310:~/video/TVShows/Mr. Robot# subliminal -l cs -v -- Mr.\ Robot\ -\ 1x02\ -\ EPS1.1_ONES-AND-ZER0ES.MPEG.mkv 
INFO     [subliminal.video] Scanning video u'Mr. Robot - 1x02 - EPS1.1_ONES-AND-ZER0ES.MPEG.mkv' in u'/i-data/cfd2a302/video/TVShows/Mr. Robot'
INFO     [subliminal.api] Listing subtitles for <Episode [u'Mr Robot', 1x2]>
INFO     [subliminal.providers] Listing subtitles with provider u'opensubtitles' and languages set([<Language [cs]>])
INFO     [subliminal.providers] Found 3 subtitles
INFO     [subliminal.providers] Skipping provider u'thesubdb': no language to search for
INFO     [subliminal.providers] Listing subtitles with provider u'podnapisi' and languages set([<Language [cs]>])
INFO     [subliminal.providers] Found 0 subtitles
INFO     [subliminal.providers] Listing subtitles with provider u'addic7ed' and languages set([<Language [cs]>])
INFO     [subliminal.providers] Found 0 subtitles
INFO     [subliminal.providers] Listing subtitles with provider u'tvsubtitles' and languages set([<Language [cs]>])
INFO     [subliminal.providers] Found 0 subtitles
INFO     [subliminal.api] Found 3 subtitles total
INFO     [subliminal.subtitle] Computed score 74 with matches set([u'episode', u'series', u'season', u'video_codec', u'year', u'hash'])
INFO     [subliminal.subtitle] Computed score 62 with matches set([u'episode', u'series', u'season', u'video_codec', u'year'])
INFO     [subliminal.subtitle] Computed score 62 with matches set([u'episode', u'series', u'season', u'video_codec', u'year'])
INFO     [subliminal.api] Downloading subtitle <OpenSubtitlesSubtitle [cs]> with score 74
INFO     [subliminal.api] Saving <OpenSubtitlesSubtitle [cs]> to u'/i-data/cfd2a302/video/TVShows/Mr. Robot/Mr. Robot - 1x02 - EPS1.1_ONES-AND-ZER0ES.MPEG.cs.srt'
1 subtitle downloaded
```
Thank you for your work!

miigotu commented:
@hlavicka do you have subtitles-multi enabled? what about embedded-all? The code in SR works 100% like the standalone version.

miigotu commented:
@neoatomic I missed adding some curly brackets when I reworked the history page. I'll merge a fix in a little while.

miigotu commented:
`It clearly requests the alpha2 flags not the alpha3. (Maybe my DB still had the old download data with the alpha2 results.?)`

matigonkas commented:
after this commit it now shows this error whenever i open a shows page (pb i believe its the code for portuguese brasilian
```
2015-07-06 14:47:12 Thread-61 :: Failed doing webui callback: Traceback (most recent call last):
  File "/home/supergonkas/SickRage/sickbeard/webserve.py", line 259, in async_call
    result = function(**kwargs)
  File "/home/supergonkas/SickRage/sickbeard/webserve.py", line 1297, in displayShow
    return t.respond()
  File "/home/supergonkas/SickRage/cache/cheetah/_home_supergonkas_SickRage_gui_slick_interfaces_default_displayShow_tmpl.py", line 1341, in respond
    for sub_lang in [babelfish.Language.fromietf(x) for x in VFN(VFFSL(SL,"epResult",True)["subtitles"],"split",False)(',') if VFFSL(SL,"epResult",True)["subtitles"]]: # generated from line 583, col 13
  File "/home/supergonkas/SickRage/lib/babelfish/language.py", line 124, in fromietf
    language = cls.fromalpha2(language_subtag)
  File "/home/supergonkas/SickRage/lib/babelfish/language.py", line 110, in fromcode
    return cls(*language_converters[converter].reverse(code))
  File "/home/supergonkas/SickRage/lib/babelfish/converters/__init__.py", line 165, in reverse
    raise LanguageReverseError(code)
LanguageReverseError: u'pb'
```

miigotu commented:
pb is not a language code, its a country code.

miigotu commented:
portuguese in alpha2 is pt, and non location specific alpha3 is por.  Opensubtitles code for portuguese brazil is pob.

miigotu commented:
the real language code for pob is pt_BR.utf-8

matigonkas commented:
but previusly to the update of sublimal it was snatching the subtitles correctly ,and showing the country flags ok
The problem i see is that the brazilian dialect isnt specified 
pt referes to pt_PT

miigotu commented:
@matigonkas use pob that is the code to specify pt brazil. allowing you to DL subs with code pb is 100% incorrect as that doesn't map to any standard code. I think that was a hackaround in the ancient subliminal when there were no alpha3 codes and there was no alpha2 code to specify it in the specification.

@neoatomic is this how compact history is supposed to look?
![image](https://cloud.githubusercontent.com/assets/390379/8534250/6536e9b2-23ef-11e5-8265-e8d0ff5e879c.png)


and detailed?

![image](https://cloud.githubusercontent.com/assets/390379/8534278/a3b3b83c-23ef-11e5-875b-a631a4917210.png)

displayshow:

![image](https://cloud.githubusercontent.com/assets/390379/8534285/c29b7f46-23ef-11e5-911d-e8d2ce0b6c29.png)


notifications still not showing the flags, but it's snatching way more now.

miigotu commented:
And this is fixed:

![image](https://cloud.githubusercontent.com/assets/390379/8534440/307753fe-23f1-11e5-8ef7-8a794b60fcf0.png)

neoatomic commented:
@miigotu Yes looks correct now.  :)

miigotu commented:
I can't make the ui notifications to work right. I think this is because of webkit updates in chrome and firefox.

matigonkas commented:
@miigotu do you know of any way ( on linux) to change in batch all the subtitles i havê from .pb.srt to .pob.srt ?

fernandog commented:
Kodi detects OK 'pt' as as Portuguese AFAIK

@matigonkas test it before change all subs

miigotu commented:
@matigonkas 

https://github.com/Diaoul/subliminal/issues/323
You can, but I wouldnt recommend it. This is a bug in subliminal.

cd to show dir and do this:

(this will show you what will change first, to be safe)
`find . -name '*.srt' -exec rename -v -n 's/.pb.srt/.pob.srt/' {} +`

then to actually do it:
`find . -name '*.srt' -exec rename -v 's/.pb.srt/.pob.srt/' {} +`

reverse:
`find . -name '*.srt' -exec rename -v 's/.pob.srt/.pb.srt/' {} +`

miigotu commented:
pt is portuguese, but not brazillian specific.

I'm not sure if kodi/vlc recognize pob, from reading that thread. It seems everyone else understands pb except subliminal (because pb is a made up code and subliminal aims to be accurate). Subliminal is working on a fix.

miigotu commented:
@neoatomic about your 2 letter codes in db, I would rename the subs to the 3 letter code, then go in subs settings and remove and readd your wanted subs languages. When SR scans your subs again it will update the sub lang with 3ltr code I think.

(This wont be an issue once the switch is made)

matigonkas commented:
@miigotu & @fernandog  i changed the subs on a show just to test fom .pb to .pob and  SR doesnt trhow an error anymore BUT it shows the sub flag as unknown (the question mark one) 
Also as my personal experience the pt code gets detected almost everywhere as the pb or pob shows as unknown 
(whouldnt it be a good idea to make it so that SR renames .pb .pob t avoid this ? there could be more people besides me having this)

miigotu commented:
@matigonkas https://github.com/SiCKRAGETV/SickRage/pull/2051#issuecomment-119022234

neoatomic commented:
@miigotu With "missed subtitle management" i did a scan for the missing subs of season 5 of Game of thrones.
Although the history page says it found all the subs for that season, only 4 files of the 10 are actually added to my show folder. : 

![2](https://cloud.githubusercontent.com/assets/7928052/8542728/eaa124ac-2497-11e5-8150-6f318cde73ab.png)
![naamloos](https://cloud.githubusercontent.com/assets/7928052/8542730/edf6d61a-2497-11e5-8bec-c617b2fc51f8.png)

Sadly noting in the Log : 
http://pastebin.com/SJxeiFXF

Also those 4 added subs are slightly out of sync, so probably podnapisi.net doesn't use a hash.? 
(maybe we can add an " * " behind the providers that use a hash.? It will help users choosing.)

miigotu commented:
check SR dir for the missing subs, I committed a fix for that tonight where the second Episode(video) object passed to subliminal had no path just a name, so subliminal saved them relative to SickBeard.py.

miigotu commented:
And yes it looks like podnapisi doesn't use hashes, I'll work on that (*) soon as I can

miigotu commented:
Also, maybe I should have left the second video object out altogether, or just copy the source/screensize/quality/rls_group from it to the first object.

The case where this is useful is when users have renamed episodes with no source/quality/group in the file name. Without this information in the filename it is harder to find a match, because guessit usually only uses the filename.

When changing something this major there is always a period of working out the kinks. (convincing myself....again)

miigotu commented:
```
>>> vars(subliminal.scan_video('../Shows/Game of Thrones/Season 05/Game of Thrones - S05E03 - High Sparrow.mp4'))
{'tvdb_id': None, 'episode': 3, 'name': '../Shows/Game of Thrones/Season 05/Game of Thrones - S05E03 - High Sparrow.mp4', 'subtitle_languages': set([]), 'format': None, 'series': u'Game of Thrones', 'title': u'High Sparrow', 'year': None, 'imdb_id': None, 'audio_codec': None, 'season': 5, 'video_codec': None, 'hashes': {u'opensubtitles': u'6f2b9faf4e742931', u'thesubdb': '95ac0d6d1c2ef55441f745a0239ae445'}, 'resolution': None, 'release_group': None, 'size': 348693872}
```

miigotu commented:
The way it works is, hash match is a score of 100, but if no hash matches then the more of the other variables the better the score. Score of 0 means no match and no sub gets downloaded.

With SR, we will always have series, episode and season from the filename. I also can always add the release_group, tvdbid and imdbid from our db.

I need to handle the cases where release_group, episode title, resolution, etc are missing to provide better matches

fernandog commented:
"I also can always add the release_group, tvdbid and imdbid from our db."

I second that. Hash works but we need fallback.
I prefer search by season and episode

miigotu commented:
@fernandog it always falls back with series, season, and episode... but that is not accurate as you get subs out of sync. Thats why we need codec/resolution/rls_group/etc to make better matches.

Its like 'ok I found 100 matches but this incorrect one is better than the real one we found because you dont have enough matches'

miigotu commented:
The solution that would work the best, which i'm not sure is a good idea or not, would be to:
1. temporarily create a hard link of the episode with the release name right next to it if it isnt already the same name
2. scan the video with the right release name
3. let subliminal save the subs there (with the same name as release name)
4. remove the hardlink and rename the sub to the name of the episode in your library

Are there people who absolutely cannot create hard links (even in the same folder)?

fernandog commented:
@miigotu it's not accurate because release group from subs **MUST** be the same as the video file.

miigotu commented:
@fernandog its not just release group, because release group can release more than one version of the same episode, in more than one format/resolution or even proper.

neoatomic commented:
@miigotu The missing subs where indeed hiding in the SR folder. :)

I prefer hash searches there for only use opensubtitels and addic7ed. 
My experience is that there are to many out of sync/differences between HDTV captures, webDL and bluray rips to relay on season/episode. 
If it was possible (with an extra setting) to enable "hash only" for podnapisi, that would make me a happy camper.

miigotu commented:
That is why I tried creating a second object to scan from, using the relase_name, but it is causing havoc.
Anyone see a reason I cannot do it with a hard link to be more accurate?

fernandog commented:
"Anyone see a reason I cannot do it with a hard link to be more accurate?" - didn't understand. sorry

miigotu commented:
@neoatomic podnapisi doesnt even have hashes as far as I can see, and addic7ed needs a password which I don't have support for yet.

@fernandog https://github.com/SiCKRAGETV/SickRage/pull/2051#issuecomment-119191809

miigotu commented:
Here is the difference of using the release name vs the name you renamed it to in your library:

```
>>> vars(subliminal.video.Episode.fromname('Game of Thrones - S05E02 - The House of Black and White.mkv'))
{'tvdb_id': None, 'episode': 2, 'name': 'Game of Thrones - S05E02 - The House of Black and White.mkv', 'subtitle_languages': set([]), 'format': None, 'series': u'Game of Thrones', 'title': u'The House of Black and White', 'year': None, 'imdb_id': None, 'audio_codec': None, 'season': 5, 'video_codec': None, 'hashes': {}, 'resolution': None, 'release_group': None, 'size': None}
>>> vars(subliminal.video.Episode.fromname('Game.of.Thrones.S05E04.HDTV.x264-Xclusive'))
{'tvdb_id': None, 'episode': 4, 'name': 'Game.of.Thrones.S05E04.HDTV.x264-Xclusive', 'subtitle_languages': set([]), 'format': u'HDTV', 'series': u'Game of Thrones', 'title': None, 'year': None, 'imdb_id': None, 'audio_codec': None, 'season': 5, 'video_codec': None, 'hashes': {}, 'resolution': None, 'release_group': None, 'size': None}
```

fernandog commented:
@miigotu can't we get the original name before renaming from history? 
Also, if users wants subs they know they must have the release name, so I won't make complex code if they don't want release name in filename. I would do like that: warning if user don't have release group in filename when subtitle is enabled. how has release group, do nothing.

fernandog commented:
strainge the 'release_group': None in the second command

miigotu commented:
(I forgot the file extension EDITTED)

You misinderstand, I can make it work with whatever they have the file named as if they have a release_name in db. Otherwise it will fallback to hash and then back to series/season/episode

`ln 'Game of Thrones - S05E04 - Sons of the Harpy.mp4' 'Game.of.Thrones.S05E04.HDTV.x264-Xclusive.mp4'`
video1 = subliminal.scan_video('Game of Thrones - S05E04 - Sons of the Harpy.mp4')
video2 = subliminal.scan_video('Game.of.Thrones.S05E04.HDTV.x264-Xclusive.mp4')
if video2.title and not video1.title:
    video1.title = video2.title

video1.imdb_id = self.show.imdbid
video1.tvdb_id = self.show.indexerid
if not video1.release_group and self.rls_group:
    video1.release_group = self.rls_group

subs = subliminal.download_best_subtitles([video1])
subliminal.save_subtitles(subs)

rm Game.of.Thrones.S05E04.HDTV.x264-Xclusive.mp4
mv Game.of.Thrones.S05E04.HDTV.x264-Xclusive.srt 'Game of Thrones - S05E04 - Sons of the Harpy.srt'

fernandog commented:
why don't you pass all the parameter from video1 to video2 and only change the name of the file in "subliminal.scan_video" . You don't need to create a hardlink. Agree?

neoatomic commented:
Not sure if this helps, but there is an option "sEH - Exact hash match (OSH) " 

https://www.podnapisi.net/forum/viewtopic.php?f=62&t=26164#p212652

miigotu commented:
@neoatomic ahh so it does use hash for posnapisi, just doesnt show the hash on their site =P

@fernandog I tried that, but the internal calculation made by subliminal is thrown off if I update things in the object, and it doesnt help. Plus the 'name' attribute has to stay the same as where the file gets saved, but it needs to be the same as the release to match an exact release.

This is simply a limitation of subliminal atm, we can ask for it as a feature to pass an 'original name' to scan_video(s), but it doesnt support it now. After a file is renamed, the accuracy of a non hash match is greatly reduced.

fernandog commented:
@miigotu so we warn user that about it. release name must be in the filename if he wants subtitle with non-hash. AS it's a fallback if there's no release name, don't fall back to season+episode

hlavicka commented:
@miigotu seems using only opensubtitles plugin does not download subtitles. I tested it with Game of Thrones S05E04 SDTV and Game of Thrones S05E10 HDTV downloaded today. I have selected Spanish, English, French, German, Italian and Czech as requested subtitles languages. I have history yes, multi-language = yes, embedded = ignore. I was able to download subtitles with other plugins. 
With subliminal cli I was able to download subtitles only using opensubtitles provider.

miigotu commented:
@fernandog we don't process if self.location is not real, so we always have a hash. I am not going to change code in subliminal to stop it from picking a subtitle if there is no hash match, instead I use this link method to make sure we have as much accurate information about the file as possible in order for subliminal to pick higher scored subtitles when the hash is not matched on any provider. This extra information will only be available though if the episode has a release_name in the db, otherwise it has to go forward using only the hash information and info from self.location filename.

abeloin commented:
@miigotu unraid doesn't support hardlink: https://github.com/SiCKRAGETV/sickrage-issues/issues/563#issuecomment-73440777

miigotu commented:
See my line comment on the PR, I catch any exception in the attempt to create the new string and hard link, and it falls back to just using self.location instead just as if self.release_name were missing.

neoatomic commented:
@miigotu Just updated.

```
2015-07-08 11:23:42 DEBUG    CHECKVERSION :: cur_commit = c1c1d19b639cf8b7957e3647ca464b3d5e67f1bb, newest_commit = c1c1d19b639cf8b7957e3647ca464b3d5e67f1bb, num_commits_behind = 0, num_commits_ahead = 0

```

```
2015-07-08 11:24:43 DEBUG    FINDSUBTITLES :: Downloading subtitles for episode 2x9 of show Penny Dreadful
2015-07-08 11:24:43 DEBUG    FINDSUBTITLES :: 265766: An object for episode 2x9 didn't exist in the cache, trying to create it
2015-07-08 11:24:43 DEBUG    FINDSUBTITLES :: 265766: Loading episode details from DB for episode 2x9
2015-07-08 11:24:43 DEBUG    FINDSUBTITLES :: Setter sets location to /volume1/video/=_Series_=/=_Nieuw_=/Penny Dreadful/Penny.Dreadful.S02E09.720p.HDTV.X264-DIMENSION.mkv
2015-07-08 11:24:43 DEBUG    FINDSUBTITLES :: 265766: Downloading subtitles for S02E09
2015-07-08 11:25:09 ERROR    FINDSUBTITLES :: Error occurred when downloading subtitles: Traceback (most recent call last):
  File "/volume1/@appstore/********/var/SickRage/sickbeard/tv.py", line 1518, in downloadSubtitles
    path = subliminal.get_subtitle_path(video.name, sub.language if sickbeard.SUBTITLES_MULTI else None)
AttributeError: 'module' object has no attribute 'get_subtitle_path'
Traceback (most recent call last):
  File "/volume1/@appstore/********/var/SickRage/sickbeard/tv.py", line 1518, in downloadSubtitles
    path = subliminal.get_subtitle_path(video.name, sub.language if sickbeard.SUBTITLES_MULTI else None)
AttributeError: 'module' object has no attribute 'get_subtitle_path'
2015-07-08 11:25:09 DEBUG    FINDSUBTITLES :: Downloading subtitles for episode 2x2 of show True Detective
2015-07-08 11:25:09 DEBUG    FINDSUBTITLES :: 270633: An object for episode 2x2 didn't exist in the cache, trying to create it
2015-07-08 11:25:09 DEBUG    FINDSUBTITLES :: 270633: Loading episode details from DB for episode 2x2
2015-07-08 11:25:09 DEBUG    FINDSUBTITLES :: Setter sets location to /volume1/video/=_Series_=/=_Nieuw_=/True Detective/True.Detective.S02E02.720p.HDTV.x264-0SEC.mkv
2015-07-08 11:25:09 DEBUG    FINDSUBTITLES :: 270633: Downloading subtitles for S02E02
2015-07-08 11:25:15 ERROR    FINDSUBTITLES :: Error occurred when downloading subtitles: Traceback (most recent call last):
  File "/volume1/@appstore/********/var/SickRage/sickbeard/tv.py", line 1518, in downloadSubtitles
    path = subliminal.get_subtitle_path(video.name, sub.language if sickbeard.SUBTITLES_MULTI else None)
AttributeError: 'module' object has no attribute 'get_subtitle_path'
Traceback (most recent call last):
  File "/volume1/@appstore/********/var/SickRage/sickbeard/tv.py", line 1518, in downloadSubtitles
    path = subliminal.get_subtitle_path(video.name, sub.language if sickbeard.SUBTITLES_MULTI else None)
AttributeError: 'module' object has no attribute 'get_subtitle_path'
2015-07-08 11:25:15 DEBUG    FINDSUBTITLES :: Downloading subtitles for episode 2x3 of show True Detective
2015-07-08 11:25:15 DEBUG    FINDSUBTITLES :: 270633: An object for episode 2x3 didn't exist in the cache, trying to create it
2015-07-08 11:25:15 DEBUG    FINDSUBTITLES :: 270633: Loading episode details from DB for episode 2x3
2015-07-08 11:25:15 DEBUG    FINDSUBTITLES :: Setter sets location to /volume1/video/=_Series_=/=_Nieuw_=/True Detective/true.detective.s02e03.720p.hdtv.x264-0sec.mkv
2015-07-08 11:25:15 DEBUG    FINDSUBTITLES :: 270633: Downloading subtitles for S02E03
2015-07-08 11:25:20 ERROR    FINDSUBTITLES :: Error occurred when downloading subtitles: Traceback (most recent call last):
  File "/volume1/@appstore/********/var/SickRage/sickbeard/tv.py", line 1518, in downloadSubtitles
    path = subliminal.get_subtitle_path(video.name, sub.language if sickbeard.SUBTITLES_MULTI else None)
AttributeError: 'module' object has no attribute 'get_subtitle_path'
Traceback (most recent call last):
  File "/volume1/@appstore/********/var/SickRage/sickbeard/tv.py", line 1518, in downloadSubtitles
    path = subliminal.get_subtitle_path(video.name, sub.language if sickbeard.SUBTITLES_MULTI else None)
AttributeError: 'module' object has no attribute 'get_subtitle_path'
2015-07-08 11:25:20 DEBUG    FINDSUBTITLES :: Downloading subtitles for episode 2x10 of show Penny Dreadful
2015-07-08 11:25:20 DEBUG    FINDSUBTITLES :: 265766: An object for episode 2x10 didn't exist in the cache, trying to create it
2015-07-08 11:25:20 DEBUG    FINDSUBTITLES :: 265766: Loading episode details from DB for episode 2x10
2015-07-08 11:25:20 DEBUG    FINDSUBTITLES :: Setter sets location to /volume1/video/=_Series_=/=_Nieuw_=/Penny Dreadful/Penny.Dreadful.S02E10.720p.HDTV.X264-DIMENSION.mkv
2015-07-08 11:25:21 DEBUG    FINDSUBTITLES :: 265766: Downloading subtitles for S02E10
2015-07-08 11:25:24 DEBUG    FINDSUBTITLES :: 265766: No subtitles found for S02E10 on any provider
2015-07-08 11:25:24 DEBUG    FINDSUBTITLES :: Downloading subtitles for episode 1x3 of show The Making of The Mob: New York
2015-07-08 11:25:24 DEBUG    FINDSUBTITLES :: 294781: An object for episode 1x3 didn't exist in the cache, trying to create it
2015-07-08 11:25:24 DEBUG    FINDSUBTITLES :: 294781: Loading episode details from DB for episode 1x3
2015-07-08 11:25:24 DEBUG    FINDSUBTITLES :: Setter sets location to /volume1/video/=_Series_=/=_Nieuw_=/The Making of The Mob New York/The.Making.Of.The.Mob.New.York.Part.3.720p.HDTV.x264-BATV.mkv
2015-07-08 11:25:24 DEBUG    FINDSUBTITLES :: 294781: Downloading subtitles for S01E03
2015-07-08 11:25:24 ERROR    FINDSUBTITLES :: Error occurred when downloading subtitles: Traceback (most recent call last):
  File "/volume1/@appstore/********/var/SickRage/sickbeard/tv.py", line 1476, in downloadSubtitles
    video = subliminal.scan_video(vname, subtitles=not force, embedded_subtitles=not sickbeard.EMBEDDED_SUBTITLES_ALL or not force)
  File "/usr/local/********/var/SickRage/lib/subliminal/video.py", line 203, in scan_video
    video = Video.fromguess(path, guessit.guess_file_info(path))
  File "/usr/local/********/var/SickRage/lib/subliminal/video.py", line 68, in fromguess
    raise ValueError('The guess must be an episode or a movie guess')
ValueError: The guess must be an episode or a movie guess
Traceback (most recent call last):
  File "/volume1/@appstore/********/var/SickRage/sickbeard/tv.py", line 1476, in downloadSubtitles
    video = subliminal.scan_video(vname, subtitles=not force, embedded_subtitles=not sickbeard.EMBEDDED_SUBTITLES_ALL or not force)
  File "/usr/local/********/var/SickRage/lib/subliminal/video.py", line 203, in scan_video
    video = Video.fromguess(path, guessit.guess_file_info(path))
  File "/usr/local/********/var/SickRage/lib/subliminal/video.py", line 68, in fromguess
    raise ValueError('The guess must be an episode or a movie guess')
ValueError: The guess must be an episode or a movie guess
2015-07-08 11:25:24 DEBUG    FINDSUBTITLES :: Downloading subtitles for episode 1x4 of show The Making of The Mob: New York
2015-07-08 11:25:24 DEBUG    FINDSUBTITLES :: 294781: An object for episode 1x4 didn't exist in the cache, trying to create it
2015-07-08 11:25:24 DEBUG    FINDSUBTITLES :: 294781: Loading episode details from DB for episode 1x4
2015-07-08 11:25:24 DEBUG    FINDSUBTITLES :: Setter sets location to /volume1/video/=_Series_=/=_Nieuw_=/The Making of The Mob New York/The.Making.Of.The.Mob.New.York.Part.4.720p.HDTV.x264-BATV.mkv
2015-07-08 11:25:24 DEBUG    FINDSUBTITLES :: 294781: Downloading subtitles for S01E04
2015-07-08 11:25:24 ERROR    FINDSUBTITLES :: Error occurred when downloading subtitles: Traceback (most recent call last):
  File "/volume1/@appstore/********/var/SickRage/sickbeard/tv.py", line 1476, in downloadSubtitles
    video = subliminal.scan_video(vname, subtitles=not force, embedded_subtitles=not sickbeard.EMBEDDED_SUBTITLES_ALL or not force)
  File "/usr/local/********/var/SickRage/lib/subliminal/video.py", line 203, in scan_video
    video = Video.fromguess(path, guessit.guess_file_info(path))
  File "/usr/local/********/var/SickRage/lib/subliminal/video.py", line 68, in fromguess
    raise ValueError('The guess must be an episode or a movie guess')
ValueError: The guess must be an episode or a movie guess
Traceback (most recent call last):
  File "/volume1/@appstore/********/var/SickRage/sickbeard/tv.py", line 1476, in downloadSubtitles
    video = subliminal.scan_video(vname, subtitles=not force, embedded_subtitles=not sickbeard.EMBEDDED_SUBTITLES_ALL or not force)
  File "/usr/local/********/var/SickRage/lib/subliminal/video.py", line 203, in scan_video
    video = Video.fromguess(path, guessit.guess_file_info(path))
  File "/usr/local/********/var/SickRage/lib/subliminal/video.py", line 68, in fromguess
    raise ValueError('The guess must be an episode or a movie guess')
ValueError: The guess must be an episode or a movie guess
2015-07-08 11:25:24 DEBUG    FINDSUBTITLES :: Downloading subtitles for episode 11x12 of show Deadliest Catch
2015-07-08 11:25:24 DEBUG    FINDSUBTITLES :: 78957: An object for episode 11x12 didn't exist in the cache, trying to create it
2015-07-08 11:25:24 DEBUG    FINDSUBTITLES :: 78957: Loading episode details from DB for episode 11x12
2015-07-08 11:25:24 DEBUG    FINDSUBTITLES :: Setter sets location to /volume1/video/=_Series_=/=_Nieuw_=/Deadliest Catch/Deadliest.Catch.S11E12.5.Year.Storm.Part.1.720p.HDTV.x264-DHD.mkv
2015-07-08 11:25:24 DEBUG    FINDSUBTITLES :: 78957: Downloading subtitles for S11E12
2015-07-08 11:25:27 DEBUG    FINDSUBTITLES :: 78957: No subtitles found for S11E12 on any provider
```

miigotu commented:
wtf, gimme a sec

miigotu commented:
The other error is grrr. I'll have to look again after you check that PR

neoatomic commented:
@miigotu One error is gone. The other is still there.

```
2015-07-08 12:15:34 DEBUG    FINDSUBTITLES :: Downloading subtitles for episode 1x3 of show The Making of The Mob: New York
2015-07-08 12:15:34 DEBUG    FINDSUBTITLES :: 294781: An object for episode 1x3 didn't exist in the cache, trying to create it
2015-07-08 12:15:34 DEBUG    FINDSUBTITLES :: 294781: Loading episode details from DB for episode 1x3
2015-07-08 12:15:34 DEBUG    FINDSUBTITLES :: Setter sets location to /volume1/video/=_Series_=/=_Nieuw_=/The Making of The Mob New York/The.Making.Of.The.Mob.New.York.Part.3.720p.HDTV.x264-BATV.mkv
2015-07-08 12:15:34 DEBUG    FINDSUBTITLES :: 294781: Downloading subtitles for S01E03
2015-07-08 12:15:34 ERROR    FINDSUBTITLES :: Error occurred when downloading subtitles: Traceback (most recent call last):
  File "/volume1/@appstore/********/var/SickRage/sickbeard/tv.py", line 1476, in downloadSubtitles
    video = subliminal.scan_video(vname, subtitles=not force, embedded_subtitles=not sickbeard.EMBEDDED_SUBTITLES_ALL or not force)
  File "/usr/local/********/var/SickRage/lib/subliminal/video.py", line 203, in scan_video
    video = Video.fromguess(path, guessit.guess_file_info(path))
  File "/usr/local/********/var/SickRage/lib/subliminal/video.py", line 68, in fromguess
    raise ValueError('The guess must be an episode or a movie guess')
ValueError: The guess must be an episode or a movie guess
Traceback (most recent call last):
  File "/volume1/@appstore/********/var/SickRage/sickbeard/tv.py", line 1476, in downloadSubtitles
    video = subliminal.scan_video(vname, subtitles=not force, embedded_subtitles=not sickbeard.EMBEDDED_SUBTITLES_ALL or not force)
  File "/usr/local/********/var/SickRage/lib/subliminal/video.py", line 203, in scan_video
    video = Video.fromguess(path, guessit.guess_file_info(path))
  File "/usr/local/********/var/SickRage/lib/subliminal/video.py", line 68, in fromguess
    raise ValueError('The guess must be an episode or a movie guess')
ValueError: The guess must be an episode or a movie guess
2015-07-08 12:15:34 DEBUG    FINDSUBTITLES :: Downloading subtitles for episode 1x4 of show The Making of The Mob: New York
2015-07-08 12:15:34 DEBUG    FINDSUBTITLES :: 294781: An object for episode 1x4 didn't exist in the cache, trying to create it
2015-07-08 12:15:34 DEBUG    FINDSUBTITLES :: 294781: Loading episode details from DB for episode 1x4
2015-07-08 12:15:34 DEBUG    FINDSUBTITLES :: Setter sets location to /volume1/video/=_Series_=/=_Nieuw_=/The Making of The Mob New York/The.Making.Of.The.Mob.New.York.Part.4.720p.HDTV.x264-BATV.mkv
2015-07-08 12:15:34 DEBUG    FINDSUBTITLES :: 294781: Downloading subtitles for S01E04
2015-07-08 12:15:34 ERROR    FINDSUBTITLES :: Error occurred when downloading subtitles: Traceback (most recent call last):
  File "/volume1/@appstore/********/var/SickRage/sickbeard/tv.py", line 1476, in downloadSubtitles
    video = subliminal.scan_video(vname, subtitles=not force, embedded_subtitles=not sickbeard.EMBEDDED_SUBTITLES_ALL or not force)
  File "/usr/local/********/var/SickRage/lib/subliminal/video.py", line 203, in scan_video
    video = Video.fromguess(path, guessit.guess_file_info(path))
  File "/usr/local/********/var/SickRage/lib/subliminal/video.py", line 68, in fromguess
    raise ValueError('The guess must be an episode or a movie guess')
ValueError: The guess must be an episode or a movie guess
Traceback (most recent call last):
  File "/volume1/@appstore/********/var/SickRage/sickbeard/tv.py", line 1476, in downloadSubtitles
    video = subliminal.scan_video(vname, subtitles=not force, embedded_subtitles=not sickbeard.EMBEDDED_SUBTITLES_ALL or not force)
  File "/usr/local/********/var/SickRage/lib/subliminal/video.py", line 203, in scan_video
    video = Video.fromguess(path, guessit.guess_file_info(path))
  File "/usr/local/********/var/SickRage/lib/subliminal/video.py", line 68, in fromguess
    raise ValueError('The guess must be an episode or a movie guess')
ValueError: The guess must be an episode or a movie guess
2015-07-08 12:15:34 DEBUG    FINDSUBTITLES :: Downloading subtitles for episode 11x12 of show Deadliest Catch
```

neoatomic commented:
@miigotu This is the show that has "part.3" "Part.4" as episode number.

miigotu commented:
Hmm, should not affect it, at least for guessit:
```
guessit.guess_file_info('The Making of The Mob New York/The.Making.Of.The.Mob.New.York.Part.4.720p.HDTV.x264-BATV.mkv')
{u'mimetype': 'video/x-matroska', u'unidentified': [u'The Making of The Mob New York'], u'videoCodec': u'h264', u'container': u'mkv', u'title': u'The Making Of The Mob New York', u'format': u'HDTV', u'releaseGroup': u'BATV', u'part': 4, u'screenSize': u'720p', u'type': u'movie'}
>>> guessit.guess_file_info('The.Making.Of.The.Mob.New.York.Part.4.720p.HDTV.x264-BATV.mkv')
{u'mimetype': 'video/x-matroska', u'videoCodec': u'h264', u'container': u'mkv', u'title': u'The Making Of The Mob New York', u'format': u'HDTV', u'releaseGroup': u'BATV', u'part': 4, u'screenSize': u'720p', u'type': u'movie'}
```

miigotu commented:
@neoatomic Just before this line:
`video = subliminal.scan_video(vname, subtitles=not force, embedded_subtitles=not sickbeard.EMBEDDED_SUBTITLES_ALL or not force)`

can you add: (make sure it is the same number of spaces indented as the `video =` line)
`logger.log(u'vname is: %s' % vname, logger.DEBUG)`

reproduce and show me.

miigotu commented:
ahaha I know why @neoatomic 
small fix.

miigotu commented:
if not len(ek.ek(os.path.splitext, tmp)[-1]):
                        tmp += ek.ek(os.path.splitext, self.location)[-1]

it thinks the file extension is .x264-BATV so it doesnt add the file extension xD.

neoatomic commented:
Added the line. Hope its correct. :

```
2015-07-08 12:46:32 DEBUG    FINDSUBTITLES :: vname is: /volume1/video/=_Series_=/=_Nieuw_=/The Making of The Mob New York/The.Making.Of.The.Mob.New.York.Part.3.720p.HDTV.x264-BATV
2015-07-08 12:46:32 ERROR    FINDSUBTITLES :: Error occurred when downloading subtitles: Traceback (most recent call last):
  File "/volume1/@appstore/********/var/SickRage/sickbeard/tv.py", line 1477, in downloadSubtitles
    video = subliminal.scan_video(vname, subtitles=not force, embedded_subtitles=not sickbeard.EMBEDDED_SUBTITLES_ALL or not force)
  File "/usr/local/********/var/SickRage/lib/subliminal/video.py", line 203, in scan_video
    video = Video.fromguess(path, guessit.guess_file_info(path))
  File "/usr/local/********/var/SickRage/lib/subliminal/video.py", line 68, in fromguess
    raise ValueError('The guess must be an episode or a movie guess')
ValueError: The guess must be an episode or a movie guess
Traceback (most recent call last):
  File "/volume1/@appstore/********/var/SickRage/sickbeard/tv.py", line 1477, in downloadSubtitles
    video = subliminal.scan_video(vname, subtitles=not force, embedded_subtitles=not sickbeard.EMBEDDED_SUBTITLES_ALL or not force)
  File "/usr/local/********/var/SickRage/lib/subliminal/video.py", line 203, in scan_video
    video = Video.fromguess(path, guessit.guess_file_info(path))
  File "/usr/local/********/var/SickRage/lib/subliminal/video.py", line 68, in fromguess
    raise ValueError('The guess must be an episode or a movie guess')
ValueError: The guess must be an episode or a movie guess
2015-07-08 12:46:32 DEBUG    FINDSUBTITLES :: Downloading subtitles for episode 1x4 of show The Making of The Mob: New York
2015-07-08 12:46:32 DEBUG    FINDSUBTITLES :: 294781: An object for episode 1x4 didn't exist in the cache, trying to create it
2015-07-08 12:46:32 DEBUG    FINDSUBTITLES :: 294781: Loading episode details from DB for episode 1x4
2015-07-08 12:46:32 DEBUG    FINDSUBTITLES :: Setter sets location to /volume1/video/=_Series_=/=_Nieuw_=/The Making of The Mob New York/The.Making.Of.The.Mob.New.York.Part.4.720p.HDTV.x264-BATV.mkv
2015-07-08 12:46:32 DEBUG    FINDSUBTITLES :: 294781: Downloading subtitles for S01E04
2015-07-08 12:46:32 DEBUG    FINDSUBTITLES :: vname is: /volume1/video/=_Series_=/=_Nieuw_=/The Making of The Mob New York/The.Making.Of.The.Mob.New.York.Part.4.720p.HDTV.x264-BATV
2015-07-08 12:46:32 ERROR    FINDSUBTITLES :: Error occurred when downloading subtitles: Traceback (most recent call last):
  File "/volume1/@appstore/********/var/SickRage/sickbeard/tv.py", line 1477, in downloadSubtitles
    video = subliminal.scan_video(vname, subtitles=not force, embedded_subtitles=not sickbeard.EMBEDDED_SUBTITLES_ALL or not force)
  File "/usr/local/********/var/SickRage/lib/subliminal/video.py", line 203, in scan_video
    video = Video.fromguess(path, guessit.guess_file_info(path))
  File "/usr/local/********/var/SickRage/lib/subliminal/video.py", line 68, in fromguess
    raise ValueError('The guess must be an episode or a movie guess')
ValueError: The guess must be an episode or a movie guess
Traceback (most recent call last):
  File "/volume1/@appstore/********/var/SickRage/sickbeard/tv.py", line 1477, in downloadSubtitles
    video = subliminal.scan_video(vname, subtitles=not force, embedded_subtitles=not sickbeard.EMBEDDED_SUBTITLES_ALL or not force)
  File "/usr/local/********/var/SickRage/lib/subliminal/video.py", line 203, in scan_video
    video = Video.fromguess(path, guessit.guess_file_info(path))
  File "/usr/local/********/var/SickRage/lib/subliminal/video.py", line 68, in fromguess
    raise ValueError('The guess must be an episode or a movie guess')
ValueError: The guess must be an episode or a movie guess
```

miigotu commented:
Perfect, just I figured it out right before you posted xD.

miigotu commented:
Really would like to get this worked out and tested good in the next hour or so, we need to merge to master so bad.

neoatomic commented:
@miigotu Sure, i'm here to help. :) 

In the old version we had some additional sub providers. Think they where custom made by some community users. Is there a way to get them added back to the new version.? 
(As i already can imagine the tickets being opened with that question.)
And login for opensubtitels and addic7ed.?

miigotu commented:
One step at a time lol. Is it working now since the last merge?

miigotu commented:
As for the extra providers, there is a way to add some, but the other ones were removed in subliminal upstream because of incomaptibility or not working at all.

miigotu commented:
@neoatomic you might wanna check that there are not extra hardlink files in 
`/volume1/video/=_Series_=/=_Nieuw_=/The Making of The Mob New York/`

Like your normal file but without extension, you can delete them, they are hard links to the ones with extensions.

neoatomic commented:
Yes there where, they had a bad extension. I removed them.

![naamloos](https://cloud.githubusercontent.com/assets/7928052/8569119/5c1ea51e-2574-11e5-8ad4-4fef83547c09.png)

miigotu commented:
It is fixed now so let me know if it is still erroring.

neoatomic commented:
@miigotu Yes, just finished running. no errors this time. :) 

```
2015-07-08 13:27:51 DEBUG    FINDSUBTITLES :: 265766: No subtitles found for S02E10 on any provider
2015-07-08 13:27:51 DEBUG    FINDSUBTITLES :: Downloading subtitles for episode 1x3 of show The Making of The Mob: New York
2015-07-08 13:27:51 DEBUG    FINDSUBTITLES :: 294781: An object for episode 1x3 didn't exist in the cache, trying to create it
2015-07-08 13:27:51 DEBUG    FINDSUBTITLES :: 294781: Loading episode details from DB for episode 1x3
2015-07-08 13:27:51 DEBUG    FINDSUBTITLES :: Setter sets location to /volume1/video/=_Series_=/=_Nieuw_=/The Making of The Mob New York/The.Making.Of.The.Mob.New.York.Part.3.720p.HDTV.x264-BATV.mkv
2015-07-08 13:27:51 DEBUG    FINDSUBTITLES :: 294781: Downloading subtitles for S01E03
2015-07-08 13:27:53 DEBUG    FINDSUBTITLES :: 294781: No subtitles found for S01E03 on any provider
2015-07-08 13:27:53 DEBUG    FINDSUBTITLES :: Downloading subtitles for episode 1x4 of show The Making of The Mob: New York
2015-07-08 13:27:53 DEBUG    FINDSUBTITLES :: 294781: An object for episode 1x4 didn't exist in the cache, trying to create it
2015-07-08 13:27:53 DEBUG    FINDSUBTITLES :: 294781: Loading episode details from DB for episode 1x4
2015-07-08 13:27:53 DEBUG    FINDSUBTITLES :: Setter sets location to /volume1/video/=_Series_=/=_Nieuw_=/The Making of The Mob New York/The.Making.Of.The.Mob.New.York.Part.4.720p.HDTV.x264-BATV.mkv
2015-07-08 13:27:53 DEBUG    FINDSUBTITLES :: 294781: Downloading subtitles for S01E04
2015-07-08 13:27:55 DEBUG    FINDSUBTITLES :: 294781: No subtitles found for S01E04 on any provider
2015-07-08 13:27:55 DEBUG    FINDSUBTITLES :: Downloading subtitles for episode 11x12 of show Deadliest Catch 
```

miigotu commented:
No subtitles found? lol i guess that "movie" doesn't have any?

miigotu commented:
As for opensubtitles password, it just limits you to 200 subtitles per day from them right now, because subliminal is not supporting user login for it yet: https://github.com/Diaoul/subliminal/issues/365

addic7ed username and pass, I will be adding in the near future, but basic function is top priority first.

neoatomic commented:
Don't think there are any subs for that show. Thats why i normaly test with game of thrones. No problems there. :)

Anyway did some more testing. : 

Done a "missed subtitle management" no errors.
Went to the history page, all looks fine. Also downloaded 2 subs automatically, and those are good.
Compact history page looks ok.
Then went to the show page and pressed "cc"  for a episode that i already had an sub for in my folder. Got the notification "no subtitle downloaded found.
Then i remove that sub and did a manual search again. This time a sub was found and added.
(So if a subs already exist SR doesn't overwrite but gives the notification "no subs found".)
However the notification shows wrong. : 
![naamloos](https://cloud.githubusercontent.com/assets/7928052/8569389/6b8318f2-2577-11e5-91b1-765b2d22583d.png)

In short it seems to be running fine. :)

neoatomic commented:
@miigotu addic7ed works for me also without username and pass so no problem there. But it where FR that users requested.

miigotu commented:
`Then i removed that sub and did a manual search again. This time a sub was found and added.
(So if a subs already exist SR doesn't overwrite but gives the notification "no subs found".)`

Because in your db it still said you had a nld sub already, and even though it DL an nld sub, when it compared old sub (nld) == (nld) new subs it sees it is the same and no new subs. I can make it not do that, but it slows it down.

As for the html in the notification, webkit has changed alot with the new html5 drm junk and a lot of html tags just don't work in it anymore. There is no real documentation about those notifications in chrome, so should I just make it say `New subtitles: Dutch, English, Blah` ?

neoatomic commented:
@miigotu Yes, "New subtitles: Dutch" etc. is fine. (or if possible also add episode number/name)
Otherwise users will open issues for not showing the notification correctly.

neoatomic commented:
@miigotu Just did a FULL scan "missed subtitle management"  for all my shows. And got some errors.
Log : http://pastebin.com/j8dqs0Gt 

```
2015-07-08 13:49:50 ERROR    Thread-57 :: Error occurred when downloading subtitles: Traceback (most recent call last):
  File "/volume1/@appstore/********/var/SickRage/sickbeard/tv.py", line 1476, in downloadSubtitles
    video = subliminal.scan_video(vname, subtitles=not force, embedded_subtitles=not sickbeard.EMBEDDED_SUBTITLES_ALL or not force)
  File "/usr/local/********/var/SickRage/lib/subliminal/video.py", line 218, in scan_video
    mkv = enzyme.MKV(f)
  File "/usr/local/********/var/SickRage/lib/enzyme/mkv.py", line 54, in __init__
    self._parse_seekhead(seek_head, segment, stream, specs)
  File "/usr/local/********/var/SickRage/lib/enzyme/mkv.py", line 61, in _parse_seekhead
    element_name = specs[element_id][1]
KeyError: None
Traceback (most recent call last):
  File "/volume1/@appstore/********/var/SickRage/sickbeard/tv.py", line 1476, in downloadSubtitles
    video = subliminal.scan_video(vname, subtitles=not force, embedded_subtitles=not sickbeard.EMBEDDED_SUBTITLES_ALL or not force)
  File "/usr/local/********/var/SickRage/lib/subliminal/video.py", line 218, in scan_video
    mkv = enzyme.MKV(f)
  File "/usr/local/********/var/SickRage/lib/enzyme/mkv.py", line 54, in __init__
    self._parse_seekhead(seek_head, segment, stream, specs)
  File "/usr/local/********/var/SickRage/lib/enzyme/mkv.py", line 61, in _parse_seekhead
    element_name = specs[element_id][1]
KeyError: None
```

miigotu commented:
Those ones with errors are not really mkv files or they are corrupted. Someone maybe renamed them to mkv because the rest in the pack were mkv? lol

either subliminal or enzyme should be catching this exception

neoatomic commented:
@miigotu Ok, need to check those files than.

Anyway here a screen shot of the history page after the full scan.
It clearly finds more subs than with the old version. :+1: 
 
![naamloos](https://cloud.githubusercontent.com/assets/7928052/8569848/ee9b9436-257b-11e5-92ca-8a3aca8be2fa.png)

fernandog commented:
@miigotu if subtitle thread crashes what happens with hardlinks created by it?

miigotu commented:
@fernandog I was just thinking about that, and until it is fixed upstream I need to try/except on the scan_video and then remove the hardlink in the exception.

fernandog commented:
did you open an issue at subliminal with the problems you have find? and the feature request about the release name so we can stop creating hard links?

miigotu commented:
I will when we get it all worked out. I can work around them in SR code without modifying the libraries for now, then FR and PR subliminal and enzyme.

fernandog commented:
in the future we should update all libs to avoid all this trouble like subtitles module. and stop doing local changes in lib.

miigotu commented:
@fernandog Yes I know, that is the proper way to do it. This trouble with subliminal is only because they rewrote the library from scratch and it was not backward compatible, so this hassle was inevitable.
